src/main/java/org/apache/sling/junit/impl/AnnotationsProcessor.java

```diff
@@ -48,19 +51,30 @@ protected void activate(ComponentContext ctx) {
     protected void deactivate(ComponentContext ctx) {
         bundleContext = null;
         log.debug("{} deactivated", this);
+        map.clear();
     }
     
     /** Process annotations on the test object */
+    @Override
     public Object process(Object testObject) throws Exception {
         log.debug("processing {}", testObject);
+        map.put(testObject, new ArrayList<>());
         for(Field f : testObject.getClass().getDeclaredFields()) {
             if(f.isAnnotationPresent(TestReference.class)) {
                 processTestReference(testObject, f);
             }
         }
         return testObject;
     }
-    
+
+    public void cleanupTest(Object test) {
+        List<ServiceGetter<?>> serviceGetters = this.map.get(test);
```

Comment:
Instead of `this.map.get(test)` you could directly write `this.map.remove(test)`. It returns the previously assigned value.